.github/workflows/python-tests.yml

```diff
@@ -0,0 +1,43 @@
+name: Python Tests for Spark
+
+on:
+  push:
+    branches:
+      - '4.0'
+      - 'master'
+  pull_request:
+    branches:
+      - '*'
+jobs:
+  test:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python-version: [ 3.6, 3.7, 3.8 ]
+        neo4j-version: [ 3.5, 4.0, 4.1, 4.2 ]
+        spark-version: [ {short: 2.4, ext: 2.4.5 }, {short: 3.0, ext: 3.0.1} ]
+
+    steps:
+      - uses: actions/checkout@v2
+      - name: Set up Python ${{ matrix.python-version }}
+        uses: actions/setup-python@v2
+        with:
+          python-version: ${{ matrix.python-version }}
+      - name: Set up JDK 8
+        if: ${{ matrix.spark-version.short == 2.4}}
+        uses: actions/setup-java@v1
+        with:
+          java-version: 1.8
+      - name: Install dependencies
+        run: |
+          python -m pip install --upgrade pip
+          pip install pypandoc
+          pip install pyspark==${{ matrix.spark-version.ext }} "testcontainers[neo4j]"
+      - name: Build artifact
+        run: |
+          ./mvnw clean package -DskipTests -P spark-${{ matrix.spark-version }} -P scala-2.12 --no-transfer-progress
+      - name: Run tests for Spark ${{ matrix.spark-version.ext }} and Neo4j ${{ matrix.neo4j-version }}
+        if: ${{ ! (matrix.spark-version.short == 2.4 && matrix.python-version == 3.8) }}
```

Comment:
@utnaf maybe makes sense add it also in other steps?

Reply:
mmm... idk, I'd rather find a cleaner solution...like having 2 jobs, one for spark 3 and one for spark 2.4

Reply:
I don't like having a situation like that except for the case we have different python scripts for `3.0` and `2.4`, but seems not the case, the problem here is that we cannot leverage the `if:` with `matrix` directly into the `build` branch, there is a feature request into the Gh Actions repo, I hope that they will fix it asap